src/spu/src/liveness_check.rs

```diff
@@ -0,0 +1,24 @@
+use std::{
+    io::Cursor,
+    io::Result,
+    net::{SocketAddr, TcpStream},
+};
+use std::io::prelude::*;
+
+use tracing::debug;
+
+use fluvio_protocol::{Encoder, api::RequestMessage};
+use dataplane::versions::ApiVersionsRequest;
+
+pub fn probe(endpoint: SocketAddr) -> Result<()> {
```

Comment:
Is checking SPU or SC?

Reply:
SPU

Reply:
As I understand, probe should exist outside application not inside

Reply:
I think what he's doing is adding the client and server side of this probe in the same executable. So the SPU server running in the pod will respond to probe requests, but the K8 cluster also needs to know how to send requests that the server will respond to. So this subcommand of the SPU executable will act as a client for that probe request. Is that right @nacardin?

Reply:
@nicholastmosher Correct